src/dndbeyond/content-scripts/character.js

```diff
@@ -1174,27 +1160,22 @@ async function rollAction(paneClass, force_to_hit_only = false, force_damages_on
         if (character.hasClassFeature("Hexblade’s Curse") &&
             character.getSetting("warlock-hexblade-curse", false))
             critical_limit = 19;
-        // Polearm master bonus attack using the other end of the polearm is considered a melee attack.
-        // check for both versions
-        if (action_name.includes("Polearm Master") && character.hasGreatWeaponFighting(2014)) {
-            damages[0] = damages[0].replace(/[0-9]*d[0-9]+/g, "$&ro<=2");
-        } else if (action_name.includes("Polearm Master") && character.hasGreatWeaponFighting(2024)) {
-            damages[0] = damages[0].replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
-                return new Array(parseInt(amount) || 1).fill(`1d${faces}${roll_mods}min3`).join(" + ") + mods;
-            });
-        }
 
-        // check for both versions
-        if (action_name.includes("Pole Strike") && (character.hasFeat("Polearm Master 2024", false)) && 
-            character.hasGreatWeaponFighting(2024)) {
-            damages[0] = damages[0].replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
-                return new Array(parseInt(amount) || 1).fill(`1d${faces}${roll_mods}min3`).join(" + ") + mods;
-            });
-        } else if (action_name.includes("Pole Strike") && (character.hasFeat("Polearm Master 2024", false)) && 
-            character.hasGreatWeaponFighting(2014)) {
-            damages[0] = damages[0].replace(/[0-9]*d[0-9]+/g, "$&ro<=2");
+        // 2024 brutal strike add extra die and apply GWF
+        if (character.getSetting("brutal-strike") && character.hasClassFeature("Brutal Strike"))  {  
+            let strikeDieCount = character.hasClassFeature("Improved Brutal Strike") ? 2 : 1; 
+
+            let brutal_strike_dmg = applyGWFIfRequired(action_name, properties, `${strikeDieCount}d10`);
+
+            damages.push(brutal_strike_dmg);
+            damage_types.push("Brutal Strike");
+
+            settings_to_change["brutal-strike"] = false;
         }
 
+        // apply for normal die
+        damages[0] = applyGWFIfRequired(action_name, properties, damages[0]);
```

Comment:
Slight difference here in the code, old code was checking for `action_name.includes("Polearm Master")` while the new code (in utils) checks for `action_name == "Polearm Master - Bonus Attack"`
Also noticed there's a check here for `&& character.hasFeat("Polearm Master 2024", false)` which disappeared from the new utils function. Are we introducing a regression here possibly for some weird use case?

Reply:
good point those should be back (the hasfeat check) the action name we should be more lenient and check if it includes polearm master